lib/toolbox/Taskfile.yaml

```diff
@@ -1,59 +1,107 @@
-      - task: install
-      - task: run
+      - aws ecr get-login-password --region {{.AWS_REGION}} | docker login --username AWS --password-stdin {{.REGISTRY_URL_BASE}}
+
+  clean:
+    desc: Cleans all toolbox images off of the local machine
+    prompt: Are you sure you want to clean all {{.TOOLBOX_NAME}} toolbox images off of the local machine?
+    requires:
+      vars:
+        - TOOLBOX_NAME
+    cmds:
+      - docker rmi $(docker images --format '{{ "{{" }}.Repository{{ "}}" }}:{{ "{{" }}.Tag{{ "}}" }}' | grep '{{.TOOLBOX_NAME}}')
 
   build:
     desc: Build and tag our toolbox image
     requires:
       vars:
-        - IMAGE_NAME
+        - TOOLBOX_NAME
         - CONTAINERFILE_PATH
+        - REGISTRY_URL_BASE
     cmds:
-      - "docker build --tag {{.LATEST_IMAGE_TAG}} --tag {{.VERSION_IMAGE_TAG}} -f {{.CONTAINERFILE_PATH}} ."
+      - |
+        docker build --tag {{.LATEST_IMAGE_TAG}} \
+                     --tag {{.VERSION_IMAGE_TAG}} \
+                     --tag {{.REGISTRY_LATEST_URL}} \
+                     --tag {{.REGISTRY_VERSION_URL}} \
+                     -f {{.CONTAINERFILE_PATH}} .
       - echo "Done building and tagging toolbox 💯 Tagged as {{.VERSION_IMAGE_TAG}} + {{.LATEST_IMAGE_TAG}}"
 
-  install:
-    desc: Install wrapper script from geodesic container
+  pull:
+    desc: Pull our toolbox image from registry
+    deps:
+      - auth
     requires:
       vars:
-        - IMAGE_NAME
+        - REGISTRY_URL_BASE
+        - REGISTRY_LATEST_URL
     cmds:
-      - docker run --rm {{.LATEST_IMAGE_TAG}} init | bash -s {{.IMAGE_NAME}} || (echo 'Try "sudo make install"'; exit 1)
+      - docker pull {{.REGISTRY_LATEST_URL}}
+      - docker tag {{.REGISTRY_LATEST_URL}} {{.LATEST_IMAGE_TAG}}
 
-  pull:
-    desc: Pull our toolbox image from registry
+  publish:
+    desc: Publish our toolbox image to the registry
+    deps:
+      - auth
     requires:
       vars:
         - REGISTRY_URL_BASE
-        - REGISTRY_URL_FULL
+        - REGISTRY_LATEST_URL
     cmds:
-      - |
-        aws ecr get-login-password --region $(AWS_REGION) | docker login --username AWS --password-stdin {{.REGISTRY_URL_BASE}}
-        docker pull {{.REGISTRY_URL_FULL}}
+      - docker push {{.REGISTRY_LATEST_URL}}
+      - docker push {{.REGISTRY_VERSION_URL}}
+
+  install:
+    desc: Execute the toolbox wrapper script from geodesic, installing a toolbox start script onto the local host
+    summary: |
+      This executes the `init | bash` geodesic command, which pipes a bash script to the host machine which installs a
+      helper script onto the host. The helper script is used to easily start / attach the toolbox image as a container
+      and properly pass all of the relevant flags to docker (i.e. exposes a port, mounts $HOME to /localhost, etc.)
+    requires:
+      vars:
+        - TOOLBOX_NAME
+        - REGISTRY_LATEST_URL
+    cmds:
+      - docker run --rm {{.REGISTRY_LATEST_URL}} init | bash -s {{.TOOLBOX_NAME}}
```

Comment:
better but ideally there could be a link to the script itself so we can review that before running it on our machines 😁

Reply:
@kevcube good idea, done 👍